samples/tacos/src/main/kotlin/com/slack/circuit/tacos/OrderTacosCircuit.kt

```diff
@@ -0,0 +1,356 @@
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.rememberVectorPainter
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.slack.circuit.runtime.CircuitUiEvent
+import com.slack.circuit.runtime.CircuitUiState
+import com.slack.circuit.runtime.Screen
+import com.slack.circuit.runtime.presenter.Presenter
+import com.slack.circuit.tacos.model.Diet
+import com.slack.circuit.tacos.model.Ingredient
+import com.slack.circuit.tacos.step.ConfirmationOrderStep
+import com.slack.circuit.tacos.step.ConfirmationUi
+import com.slack.circuit.tacos.step.FillingsOrderStep
+import com.slack.circuit.tacos.step.FillingsUi
+import com.slack.circuit.tacos.step.OrderStep
+import com.slack.circuit.tacos.step.SummaryOrderStep
+import com.slack.circuit.tacos.step.SummaryUi
+import com.slack.circuit.tacos.step.ToppingsOrderStep
+import com.slack.circuit.tacos.step.ToppingsUi
+import com.slack.circuit.tacos.ui.theme.TacoTheme
+import java.math.BigDecimal
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.persistentSetOf
+import kotlinx.collections.immutable.toImmutableSet
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+object OrderTacosScreen : Screen {
+  data class State(
+    @StringRes val headerResId: Int,
+    val orderCost: BigDecimal,
+    val stepState: OrderStep.State,
+    val isPreviousVisible: Boolean,
+    val isNextEnabled: Boolean,
+    val isNextVisible: Boolean,
+    val isFooterVisible: Boolean,
+    val eventSink: (Event) -> Unit
+  ) : CircuitUiState
+
+  sealed interface Event : CircuitUiEvent {
+    val indexModifier: Int
+      get() = 0
+
+    object Previous : Event {
+      override val indexModifier: Int = -1
+    }
+    object Next : Event {
+      override val indexModifier: Int = 1
+    }
+    object ProcessOrder : Event {
+      override val indexModifier: Int = 1
+    }
+  }
+}
+
+@Immutable
+data class OrderDetails(
+  val filling: Ingredient = Ingredient(""),
+  val toppings: Set<Ingredient> = persistentSetOf(),
+  val baseCost: BigDecimal = BigDecimal("9.99"), // Default taco price
+  val ingredientsCost: BigDecimal = BigDecimal.ZERO
+)
+
+/** List of wizard steps (in order!) */
+private val orderSteps =
+  persistentListOf(FillingsOrderStep, ToppingsOrderStep, ConfirmationOrderStep, SummaryOrderStep)
+
+internal class OrderTacosPresenter(
+  private val fillingsProducer: OrderStep.StateProducer<FillingsOrderStep.State>,
+  private val toppingsProducer: OrderStep.StateProducer<ToppingsOrderStep.State>,
+  private val confirmationProducer: OrderStep.StateProducer<ConfirmationOrderStep.OrderState>,
+  private val summaryProducer: OrderStep.StateProducer<SummaryOrderStep.SummaryState>,
```

Comment:
This isn't really doing a composite presenter since these aren't presenters. Can we make them presenters instead for the sake of this example?

Reply:
> This isn't really doing a composite presenter since these aren't presenters

I disagree here. In my view this _is_ a composite presenter. It's composed of other objects that generate the state it needs before it can emit its own state.

When I started I assumed I would use Circuit Presenters here, however, I quickly realized that wasn't necessary or really appropriate.

These state producers aren't standalone circuits. They aren't displayed elsewhere in the application and are completely unknown to the framework. While I _could_ have them implement Circuit's Presenter interface, there's no real benefit. Doing so would further complicate the code as I'd have to create a Presenter factory for each one and inject that into `OrderTacosPresenter` so that it can provide an event sink when calling them.

Reply:
hmmm. I don't disagree, but I do also think it's sort of acting as a different kind of sample rather than being a literal composite presenter. Maybe we should have another example for that separately with a simpler case, like an index/detail flow where it's a composite presenter in tablet and multi-screen in phone. In short - I think it's more of a _complex_ (but not complicated!) presenter but not strictly a _composite_ presenter (i.e. literally composed of multiple reusable nested presenters) in the way I thought we were talking about before. I think both are useful 👍.